module/zfs/dmu_send.c

```diff
@@ -2849,9 +2860,19 @@ receive_object_range(struct receive_writer_arg *rwa,
 		return (ret);
 	}
 
-	dmu_buf_will_change_crypt_params(db, tx);
+	/*
+	 * Convert the buffer associated with this range of dnodes to a
+	 * raw buffer. This ensures that it will be written out as a raw
+	 * buffer when we fill in the dnode objects in future records.
+	 * Since we are commiting this tx now, it is technically possible
+	 * for the dnode block to end up on-disk with the incorrect MAC.
+	 * Despite this, the dataset is maked as inconsistent so no other
```

Comment:
It would be really nice to test making the receive fail when there's an incorrect MAC.  You could probably do that by receiving a send stream that's been intentionally truncated.

typo: `maked`